### PR TITLE
Adding wheel builder (currently running for every push)

### PR DESCRIPTION
**Pull request by ThomasLecocq**
### What does this PR do?

Testing the https://github.com/pypa/cibuildwheel wheel builder for all platforms, replaces #2783 

### Why was it initiated?  Any relevant Issues?

Should make our life easier... hopefully.

### CODE REVIEW

**Comment by ThomasLecocq** [comment]
OK, this recipe successfully created: 
image

I currently restricted the build to:

- 64 bits only
- python 3.9 only
- CPython only

all those other configurations are theoretically possible (but of course add a lot of build time):
image

**Comment by ThomasLecocq** [comment]
reason: I needed to test it in a PR :-) indeed :-)

**Comment by trichter** [comment]
Maybe we can upload (some) wheels to testpypi on every push to master. All wheels we want to provide can be uploaded to pypi on  a new release. Finally we might want to upload all wheels to testpypi in dedicated PRs tagged by a label `test wheel`.

**Comment by trichter** [comment]
Bye the way, who has credentials for pypi and testpypi and can add tokens for both to github?

**Comment by ThomasLecocq** [comment]
> Maybe we can upload (some) wheels to testpypi on every push to master. All wheels we want to provide can be uploaded to pypi on a new release. Finally we might want to upload all wheels to testpypi in dedicated PRs tagged by a label `test wheel`.

good idea!

**Comment by ThomasLecocq** [comment]
> Bye the way, who has credentials for pypi and testpypi and can add tokens for both to github?

I've asked Tobi, he will provide them

**Comment by trichter** [comment]
For which platforms do we want to provide wheels? Do we need aarch64, pc64le, s390x, windows32bit?
Pypi and testpypi tokens are needed to finalize this PR.

**Comment by QuLogic** [comment]
Of those, I think NumPy only provides aarch64 for recent versions.

**Comment by trichter** [comment]
@ThomasLecocq We could use testpypi to automatically upload the source distribution and wheels for tags, e.g. '1.3.0rc1'.

**Comment by megies** [comment]
Nice work on this y'all. I just saw that pypi offers 2FA and API tokens for uploading these days. Good to see, enabled F2A right away. Those plain text account credentials in `.pypirc` always felt very weird.

**Comment by megies** [comment]
I also just uploaded an old RC to testpypi to get the project set up there, added you to it too, so you can set up an API token for pypitest in the future if you wish @trichter 
https://test.pypi.org/project/obspy/

@barsch if you want to get added to obspy on testpypi you just need to verify your email, can't add you right now

**Comment by megies** [comment]
> @ThomasLecocq We could use testpypi to automatically upload the source distribution and wheels for tags, e.g. '1.3.0rc1'.

Sounds good to me, would save manually building source dist zipball and uploading for others to test

File: .github/workflows/build_wheels.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+      - name: Build sdist
+        shell: bash -l {0}
+        run: pipx run build --sdist
+
+      - uses: actions/upload-artifact@v2
+        with:
+          path: dist/*.tar.gz
+
+  publish_test:
```
Comment (d-chambers) [inline .github/workflows/build_wheels.yml:83]: Just to be clear, do you want all tags and releases to also publish to test pypi or only if the branch starts with 'test_wheel' ? If the latter we need to add more to this conditional.

**Review by d-chambers — APPROVED** [review]

File: .github/workflows/build_wheels.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
+name: wheels
+on:
+  push:
+    branches:
+      - master
+      - 'test_wheels*'
```
[the remark this answers is not in the document]
Comment (trichter) [reply .github/workflows/build_wheels.yml:6]: Exactly, I think there is some value. It allows to test the building of all wheels by pushing to a branch with name `test_wheels*`.

**Comment by trichter** [comment]
@d-chambers Thanks for the review @ThomasLecocq Thanks for the initiative, sorry for capturing your PR :wink:

**Pull request merged**